addons/website_sale_wallet/models/partner.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+
+from odoo import fields, models, api
+
+
+class Partner(models.Model):
+    _inherit = "res.partner"
+
+    wallet_ids = fields.One2many("wallet", "partner_id")
+
+    current_website = fields.Many2one('website', compute='_compute_current_website', store=False)
+    current_wallet_id = fields.Many2one("wallet", compute="_compute_default_wallet_id", store=False)
```

Comment:
because wallet is website_dependent, it should be only queryable with request.website or become a function
see: last_website_so_id field